src/main/scala/chisel3/tester/TreadleBackend.scala

```diff
@@ -237,68 +245,48 @@ object TreadleExecutive {
     }.toMap
   }
 
-  def start[T <: MultiIOModule](
-      dutGen: () => T,
-      testOptions: TesterOptions,
-      execOptions: Option[ExecutionOptionsManager] = None): BackendInstance[T] = {
-    // Create the base options manager that has all the components we care about, and initialize defaults
-    val optionsManager = new ExecutionOptionsManager("chisel3")
-        with HasChiselExecutionOptions with HasFirrtlOptions with HasTreadleSuite
-
-    // If the user specified options, override the default fields.
-    // Note: commonOptions and firrtlOptions are part of every ExecutionOptionsManager, so will always be defined
-    // whether the user intended to or not. In those cases testers2 forces an override to the testers2 defaults.
-    execOptions foreach {
-      case userOptions: HasChiselExecutionOptions => optionsManager.chiselOptions = userOptions.chiselOptions
-      case _ =>
-    }
+  def start[T <: MultiIOModule](dutGen: () => T, annotationSeq: AnnotationSeq): BackendInstance[T] = {
 
-    execOptions foreach {
-      case userOptions: HasFirrtlOptions => optionsManager.firrtlOptions = userOptions.firrtlOptions
-      case _ =>
-    }
-    optionsManager.firrtlOptions = optionsManager.firrtlOptions.copy(compilerName = "low")
+    // Force a cleanup: long SBT runs tend to fail with memory issues
+    System.gc()
 
-    execOptions foreach {
-      case userOptions: HasTreadleSuite => optionsManager.treadleOptions = userOptions.treadleOptions
-      case _ =>
-    }
+    try {
+      val generatorAnnotation = chisel3.stage.ChiselGeneratorAnnotation(dutGen)
 
-    // Tester options take priority over exec options
-    val testName = testOptions.name.replaceAll(" ", "_").replaceAll("\\W+", "")  // sanitize filename
-    optionsManager.commonOptions = optionsManager.commonOptions.copy(
-        targetDirName = s"test_run_dir/$testName")
-    if (testOptions.writeVcd) {
-      optionsManager.treadleOptions = optionsManager.treadleOptions.copy(writeVCD = true)
-    }
+      val chiseledAnnotations = (new ChiselStage).run(
+        annotationSeq ++ Seq(generatorAnnotation, NoRunFirrtlCompilerAnnotation)
+//        annotationSeq ++ Seq(generatorAnnotation)
+      )
 
-    // Force a cleanup: long SBT runs tend to fail with memory issues
-    System.gc()
+      val circuit = generatorAnnotation.elaborate.circuit
```

Comment:
Is this dependent on the post-ChiselStage annotation? In other words, is this annotation mutable?

Reply:
This annotation is not mutable.

Reply:
Can you move this above the ChiselStage invocation so that it's clear that it doesn't depend on ChiselStage running?

Reply:
moved

Reply:
Maybe I'm just using GitHub wrong but I don't see the change. Did you push your code?